packages/cubejs-redshift-driver/driver/RedshiftDriver.js

```diff
@@ -0,0 +1,82 @@
+const Redshift = require('aws-sdk/clients/redshift');
+const pg = require('pg');
+const PostgresDriver = require('@cubejs-backend/postgres-driver');
+
+const { Pool } = pg;
+
+class RedshiftDriver extends PostgresDriver {
+  constructor(config) {
+    super(config);
+    this.config = config || {};
+  }
+
+  async getCredentialsFromAWS() {
+    const [clusterIdentifier] = process.env.CUBEJS_DB_HOST && process.env.CUBEJS_DB_HOST.split('.');
+    let credentials;
+
+    try {
+      credentials = await (new Redshift()).getClusterCredentials({
+        ClusterIdentifier: clusterIdentifier,
+        DbUser: process.env.CUBEJS_DB_USER,
+      }).promise();
+    } catch (error) {
+      console.log(`An error occurred while trying to retrieve Redshift credentials: ${error.stack || error}`);
+    }
+
+    if (!credentials || !credentials.DbUser && !credentials.DbPassword) {
+      throw new Error('Unable to retrieve Redshift credentials from AWS');
+    }
+
+    return {
+      user: credentials.DbUser,
+      password: credentials.DbPassword
+    };
+  }
+
+  getCredentialsFromEnv() {
+    return {
+      user: process.env.CUBEJS_DB_USER,
+      password: process.env.CUBEJS_DB_PASS,
+    };
+  }
+
+  async createPool() {
+    if (this.pool && this.pool instanceof Pool) {
+      return;
```

Comment:
Just realized we need mutex here that will ensure only one pool is creating at the same time in case multiple queries arrived at the same time.

Reply:
Is there an example somewhere else in the codebase?

Reply:
Yep. https://github.com/cube-js/cube.js/blob/26e5843de6a844613ef6e10f0a061c84200c57b3/packages/cubejs-server-core/core/index.js#L321